components/Tabs/tests/unit/TabBarView/MDCTabBarViewTests.m

```diff
@@ -163,4 +163,59 @@ - (void)testSettingBackgroundColorUpdatesBarTintColor {
   XCTAssertEqual(self.tabBarView.barTintColor, self.tabBarView.backgroundColor);
 }
 
+- (void)testTitleColorForStateFallsBackToNormalState {
+  // Given
+  [self.tabBarView setTitleColor:nil forState:UIControlStateNormal];
+  [self.tabBarView setTitleColor:nil forState:UIControlStateSelected];
+
+  // When
+  [self.tabBarView setTitleColor:UIColor.purpleColor forState:UIControlStateNormal];
+
+  // Then
+  XCTAssertEqualObjects([self.tabBarView titleColorForState:UIControlStateSelected],
+                        UIColor.purpleColor);
+}
+
+- (void)testTitleColorForStateReturnsExpectedValue {
+  // Given
+  [self.tabBarView setTitleColor:nil forState:UIControlStateNormal];
+  [self.tabBarView setTitleColor:nil forState:UIControlStateSelected];
+
+  // When
+  [self.tabBarView setTitleColor:UIColor.purpleColor forState:UIControlStateNormal];
+  [self.tabBarView setTitleColor:UIColor.orangeColor forState:UIControlStateSelected];
+
+  // Then
+  XCTAssertEqualObjects([self.tabBarView titleColorForState:UIControlStateNormal],
+                        UIColor.purpleColor);
+  XCTAssertEqualObjects([self.tabBarView titleColorForState:UIControlStateSelected],
+                        UIColor.orangeColor);
+}
+
+- (void)testTitleColorForStateSetToNilFallsBackToNormal {
+  // Given
+  [self.tabBarView setTitleColor:nil forState:UIControlStateNormal];
+  [self.tabBarView setTitleColor:UIColor.cyanColor forState:UIControlStateSelected];
+
+  // When
+  [self.tabBarView setTitleColor:UIColor.purpleColor forState:UIControlStateNormal];
+  [self.tabBarView setTitleColor:nil forState:UIControlStateSelected];
+
+  // Then
+  XCTAssertEqualObjects([self.tabBarView titleColorForState:UIControlStateNormal],
+                        UIColor.purpleColor);
+  XCTAssertEqualObjects([self.tabBarView titleColorForState:UIControlStateSelected],
+                        UIColor.purpleColor);
+}
+
+- (void)testTitleColorForStateWithNoValuesReturnsNil {
+  // When
+  [self.tabBarView setTitleColor:nil forState:UIControlStateNormal];
+  [self.tabBarView setTitleColor:nil forState:UIControlStateSelected];
+
+  // Then
+  XCTAssertNil([self.tabBarView titleColorForState:UIControlStateNormal]);
+  XCTAssertNil([self.tabBarView titleColorForState:UIControlStateSelected]);
```

Comment:
What is the expected behavior if a title color is set for a non-normal control state, but the title color for `UIControlStateNormal` is `nil`?

Reply:
"A default color is returned."